### PR TITLE
Fixed backward compatibility for `send_email_when_published`

### PR DESCRIPTION
**Pull request by allouis**
no-issue 

refs #12343 

This restores the functionality of `send_email_when_published` query param to the Admin API's Post Edit endpoint, as well as restoring the field in both the Content & Admin API, and patching in support for the `fields` param.

Also includes a migration to drop the old column, now that it's no longer used.

### CODE REVIEW

**Comment by allouis** [comment]
@daniellockyer @rishabhgrg @kevinansfield The migration needs some :eyes: https://github.com/TryGhost/Ghost/pull/12357/commits/0d932b4c19072566d82d0e692484d4c7cf0ecda2

File: core/server/data/migrations/versions/3.38/07-remove-send-email-when-published-column.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    async down({connection}) {
+        logging.info('Adding column "send_email_when_published" to table "posts"');
+        await addColumn('posts', 'send_email_when_published', connection, {
+            type: 'bool',
+            nullable: true,
+            defaultTo: false
+        });
+
+        const trx = await connection.transaction();
```
Comment (allouis) [inline core/server/data/migrations/versions/3.38/07-remove-send-email-when-published-column.js:18]: Maybe this should be a separate migration? :thinking:

Comment (kevinansfield) [reply core/server/data/migrations/versions/3.38/07-remove-send-email-when-published-column.js:18]: We've always kept DDL queries separate from DML before. I suppose having it in a separate migration allows for separating db state between migrations but not sure how much that actually buys us 🤔

Comment (kevinansfield) [reply core/server/data/migrations/versions/3.38/07-remove-send-email-when-published-column.js:18]: This is only in the `down` command. If we perform a rollback and there's an error when migrating down do we then do a rollforward to get back to the last "good" state?

Comment (allouis) [reply core/server/data/migrations/versions/3.38/07-remove-send-email-when-published-column.js:18]: Hmmm, a roll forward would completely drop the column though, which is the last "good" state - it's a weird one!

**Comment by allouis** [comment]
Hey @daniellockyer I'm not sure why the migrations test is failing :thinking: I can't replicate locally with either MySQL or SQLite!

The error seems to point to a missing column which is weird :raised_eyebrow: 
```
update `posts` set `send_email_when_published` = true where not `email_recipient_filter` = 'none' - SQLITE_ERROR: no such column: email_recipient_filter
```

**Comment by daniellockyer** [comment]
@allouis Would you be able to rebase on top of `master`? I'm also investigating this elsewhere, but my latest commit should fix it for now 🙂

File: core/server/data/migrations/versions/3.38/07-remove-send-email-when-published-column.js (hunk 1 above)
Comment (kevinansfield) [reply core/server/data/migrations/versions/3.38/07-remove-send-email-when-published-column.js:18]: I think I'd lean more towards following existing patterns and keeping the migrations separate until we have a bigger discussion around the implications?

From the `knex-migrator` usage docs:
> Don't mix DDL/DML statements in a migration script. In MySQL DDL statements use implicit commits.

Keeping them separate also reduces cognitive overhead when deciding what to put into a migration 🤔 If there's no good reason any more to keep them separate then we should think about updating the usage docs and making sure everyone knows when it's ok to mix DDL/DML.

Comment (allouis) [reply core/server/data/migrations/versions/3.38/07-remove-send-email-when-published-column.js:18]: Cool - sounds good to me :relaxed:

Comment (allouis) [reply core/server/data/migrations/versions/3.38/07-remove-send-email-when-published-column.js]: I've updated this in the latest here: https://github.com/TryGhost/Ghost/pull/12357/commits/2bb89e9dae655412cdf97e1e81feb5a707443bdd

**Review by kevinansfield — APPROVED** [review]

**Pull request merged**